userspace/libsinsp/utils.cpp

```diff
@@ -665,56 +665,49 @@ void copy_and_sanitize_path(char* target, char* targetbase, const char* path, ch
 		}
 		else
 		{
-			if(*pc == '.' && *(pc + 1) == '.' && *(pc + 2) == separator)
+			if(*pc == '.')
```

Comment:
We may also address other corner cases we talked about @araujof:

1. `/testdir/test../testfile` would probably be sanitized incorrectly as `/testfile`
2. `/testdir/test./testfile` would probably be sanitized incorrectly as `/testdir/testtestfile`

So I think we just need to check if either the `.` is the first char in the path or the previous character is a separator prior to apply any logic. Otherwise, we can just fall in the final else case and copy the character.

```suggestion
			if(*pc == '.' && (tc == targetbase || *(tc - 1) == separator)) // handle '.' only as first char or after a `/`
```